### PR TITLE
GEODE-2668: Better handling of configuration updates

### PR DESCRIPTION
**Pull request by pdxrunner**
In cases where destroying GWRs may fail on some or all servers, control
configuration updating based on state of --ifExists option.
--ifExists == true --> update config
--ifExists == false --> do not update config

Added tests for edge cases with failures on servers.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [X] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [X] Does `gradlew build` run cleanly?

- [X] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java [hunk 1]
```diff
@@ -68,7 +68,9 @@ public Result destroyGatewayReceiver(
     // Only update the cluster config if the command is not executed on specific members.
     InternalClusterConfigurationService service =
         (InternalClusterConfigurationService) getConfigurationService();
-    if (onMember == null && result.getStatus().equals(Result.Status.OK) && service != null) {
+    if (onMember != null || service == null) {
+      persisted = false;
+    } else if (result.getStatus().equals(Result.Status.OK)/* && service != null */) {
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java]: This check should remain in place.

Comment (pdxrunner) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java]: Good catch. Left over from some debug iterations. I'll fix this along with the conflicts resulting from renaming `ClusterConfigurationService`.
